upstairs/src/lib.rs

```diff
@@ -266,19 +266,23 @@ async fn io_completed(
     data: Option<Bytes>,
     ds_done_tx: mpsc::Sender<u64>,
 ) -> Result<()> {
+    /*
+     * Having to put scope around the ds_work.lock call to avoid the
+     * compiler complaining (error[E0277]) about holding the lock while
+     * calling send (even if we drop) then creates a warning for the
+     * gw_work_done variable.
+     */
+    #[allow(unused_assignments)]
```

Comment:
to avoid that warning, this could be:
```
let mut gw_work_done = {
  let mut work = up.ds_work.lock().unwrap();
  work.complete(ds_id, client_id, data)?
};
```

Reply:
Fixed in 960fc31